File: tests/ten_runtime/integration/go/start_graph_and_set_dests_from_new_graph_go/start_graph_and_set_dests_from_new_graph_go_app/ten_packages/extension/test_extension_3/extension.go
//
// Copyright © 2025 Agora
// This file is part of TEN Framework, an open source project.
// Licensed under the Apache License, Version 2.0, with certain conditions.
// Refer to the "LICENSE" file in the root directory for more information.
//

package test_extension_3

import (
	ten "ten_framework/ten_runtime"
)

type testExtension3 struct {
	ten.DefaultExtension
}

func (ext *testExtension3) OnCmd(tenEnv ten.TenEnv, cmd ten.Cmd) {
	cmdName, _ := cmd.GetName()

	if cmdName == "B" {
		ext.handleBCmd(tenEnv, cmd)
	} else {
		panic("test_extension_3 received unexpected cmd: " + cmdName)
	}
}

func (ext *testExtension3) handleBCmd(tenEnv ten.TenEnv, cmd ten.Cmd) {
	// Simply return OK status for command B
	result, _ := ten.NewCmdResult(ten.StatusCodeOk, cmd)
	tenEnv.ReturnResult(result, nil)
}

func newTestExtension3(name string) ten.Extension {
	return &testExtension3{}
}

func init() {
	err := ten.RegisterAddonAsExtension(
		"test_extension_3",
		ten.NewDefaultExtensionAddon(newTestExtension3),
	)
	if err != nil {
		panic("Failed to register addon: " + err.Error())
	}
}

File: tests/ten_runtime/integration/python/start_graph_and_set_dests_from_new_graph_python/start_graph_and_set_dests_from_new_graph_python_app/ten_packages/extension/test_extension_3/extension.py
#
# Copyright © 2025 Agora
# This file is part of TEN Framework, an open source project.
# Licensed under the Apache License, Version 2.0, with certain conditions.
# Refer to the "LICENSE" file in the root directory for more information.
#

from ten_runtime import (
    Extension,
    TenEnv,
    Cmd,
    StatusCode,
    CmdResult,
    LogLevel,
)


class TestExtension3(Extension):
    def __init__(self, name: str) -> None:
        super().__init__(name)

    def on_cmd(self, ten_env: TenEnv, cmd: Cmd) -> None:
        cmd_name = cmd.get_name()

        if cmd_name == "B":
            self._handle_b_cmd(ten_env, cmd)
        else:
            ten_env.log(
                LogLevel.ERROR,
                f"test_extension_3 received unexpected cmd: {cmd_name}",
            )

    def _handle_b_cmd(self, ten_env: TenEnv, cmd: Cmd) -> None:
        # Simply return OK status for command B
        result = CmdResult.create(StatusCode.OK, cmd)
        ten_env.return_result(result)
